text/xxxx-metrics-correlation.md

```diff
@@ -0,0 +1,140 @@
+- Start Date: YYYY-MM-DD
+- RFC Type: feature
+- RFC PR: <link>
+- RFC Status: draft
+
+# Summary
+
+This RFC addresses the high level metrics to span correlation system.
+
+# Motivation
+
+We believe the value in a good metrics solution is the correlation to traces and other signals.
+This means that we need to store evidence of metric measurements on the spans.  These
+measurements are automatically added when using the basic `metrics` API.
+
+# Basics
+
+Whenever a metrics API is used it operates in either span seeking or span creating mode.  Most
+of the metrics APIs are span seeking which means that they record a measurement in relation
+to that span.  Some APIs (such as `metrics.timing` when used with a code block) will instead
+create a span and bind it.
+
+```python
+def process_batch(batch):
+    processor = Processor()
+
+    # This creates a span with op `timing_measurement`
+    with metrics.timing("processor.process_batch"):
```

Comment:
Let's use the naming scheme of span name/metric name instead of using `op` (aligns with otel, removes educating about op types, aligns with other metrics products).

Op already has a meaning to sentry today (`db.X`, `resource.Y`), and that concept generally has caused us issues, let's not introduce it for any metrics API (we can always make op an optional thing people can set and map it to the same semantics as span op).

Reply:
I mostly wrote this up in the light of what we have today. I agree that the op/description situation is not great so ideally this can recommend something else instead.